src/services/permissions-getter.js

```diff
@@ -0,0 +1,193 @@
+    const newPermissions = {};
+
+    Object.keys(permissions).forEach((modelName) => {
+      const modelPermissions = permissions[modelName];
+      const { collection } = modelPermissions;
+
+      newPermissions[modelName] = {
+        collection: {
+          browseEnabled: collection.list || collection.searchToEdit,
+          readEnabled: collection.show,
+          editEnabled: collection.update,
+          addEnabled: collection.create,
+          deleteEnabled: collection.delete,
+          exportEnabled: collection.export,
+        },
+        scope: modelPermissions.scope,
+      };
+
+      if (modelPermissions.actions) {
+        newPermissions[modelName].actions = PermissionsGetter
+          ._transformActionsPermissionsFromOldToNewFormat(modelPermissions.actions);
+      }
+    });
+
+    return newPermissions;
+  }
+
+  static _setTeamsACLPermissions(renderingId, permissions) {
+    const newFormatPermissions = permissions
+      ? PermissionsGetter._transformPermissionsFromOldToNewFormat(permissions)
+      : null;
+
+    if (!PermissionsGetter.permissions.renderings) {
+      PermissionsGetter.permissions.renderings = {};
+    }
+    PermissionsGetter.permissions.renderings[renderingId] = {
+      data: newFormatPermissions,
+      lastRetrieve: moment(),
+    };
+  }
+
+  static _setRolesACLPermissions(renderingId, permissions) {
+    PermissionsGetter.permissions.collections = {
+      data: permissions.collections,
+      lastRetrieve: moment(),
+    };
+
+    if (!PermissionsGetter.permissions.renderings) {
+      PermissionsGetter.permissions.renderings = {};
+    }
+    PermissionsGetter.permissions.renderings[renderingId] = {
+      data: permissions.renderings ? permissions.renderings[renderingId] : null,
+      lastRetrieve: moment(),
+    };
+  }
+
+  // In the teamACL format, all the permissions are stored by renderingId into "renderings".
+  // For the rolesACL format, the collections permissions are stored directly into "collections",
+  // and only their scopes are stored by renderingId into "renderings".
+  static _setPermissions(renderingId, permissions) {
+    if (PermissionsGetter.isRolesACLActivated) {
+      PermissionsGetter._setRolesACLPermissions(renderingId, permissions);
+    } else {
+      PermissionsGetter._setTeamsACLPermissions(renderingId, permissions);
+    }
+  }
+
+  static getLastRetrieveTime(renderingId, permissionName) {
+    // In the case of rolesACL format and browseEnabled permission, the last retrieve to be taken
+    // into account is the one stored by rendering (because of the scope information).
+    if (PermissionsGetter.isRolesACLActivated && PermissionsGetter._getPermissionsInCollections() && permissionName !== 'browseEnabled') {
+      return PermissionsGetter._getPermissionsInCollections().lastRetrieve;
+    }
+    return PermissionsGetter._getPermissionsInRendering(renderingId)
+      ? PermissionsGetter._getPermissionsInRendering(renderingId).lastRetrieve
+      : null;
```

Comment:
You have an issue ere, but not a big one. I'll talk about it in your next PR

Reply:
This function will disappear in the next PR anyway 😉  But sure let's talk about it whenever you want!

Reply:
Oh I think I know what you mean. It concerns the "browseEnabled" case when you can have retrieve only on renderings?
If it is, don't worry I take care of this in the next PR 👍 
In this PR, for rolesACL, the lastRetrieve are the same in collectionPermissions and in renderingPermissions.